### PR TITLE
ATO-1182: Refactor sessions setting in AuthenticationCallback

### PR DESCRIPTION
**Pull request by billrdunn**
## What

While working on setting auth_time in AuthenticationCallback, it was becoming a bit complex and hard to read. For maintainability, this PR contains two small refactors:

1. Where possible, separate setting to the session objects, and updating the database.
2. Use private method to separate logic from main handler.

## Testing
No changes to tests required, so I'm confident this is OK, but I will

- [x] Run through auth journey in dev

### CODE REVIEW

File: oidc-api/src/main/java/uk/gov/di/authentication/oidc/lambda/AuthenticationCallbackHandler.java [hunk 1]
```diff
@@ -396,10 +396,13 @@ public APIGatewayProxyResponseEvent handleRequest(
                                     ? OrchSessionItem.AccountState.NEW
                                     : OrchSessionItem.AccountState.EXISTING;
                 }
-                sessionService.storeOrUpdateSession(userSession.setNewAccount(accountState));
-                orchSessionService.updateSession(orchSession.withAccountState(orchAccountState));
+                userSession = userSession.setNewAccount(accountState);
+                orchSession = orchSession.withAccountState(orchAccountState);
 
-                sessionService.storeOrUpdateSession(userSession.setAuthenticated(true));
+                userSession = userSession.setAuthenticated(true);
```
Comment (GHSwallow) [inline oidc-api/src/main/java/uk/gov/di/authentication/oidc/lambda/AuthenticationCallbackHandler.java]: The last commit is making a change to lines changed in the second commit. I think these lines refactored in the final commit can just be refactored in the second commit instead

Comment (billrdunn) [reply oidc-api/src/main/java/uk/gov/di/authentication/oidc/lambda/AuthenticationCallbackHandler.java]: Fixed :)

File: oidc-api/src/test/java/uk/gov/di/authentication/oidc/lambda/AuthenticationCallbackHandlerTest.java [hunk 2]
```diff
@@ -380,6 +380,9 @@ void shouldRedirectToFrontendErrorPageIfUserInfoRequestIsUnsuccessful()
     @Test
     void shouldUpdateOrchSessionUsingClaimsFromUserInfoResponse()
             throws UnsuccessfulCredentialResponseException {
+
+        when(USER_INFO.getBooleanClaim("new_account")).thenReturn(true);
+
```
Comment (GHSwallow) [inline oidc-api/src/test/java/uk/gov/di/authentication/oidc/lambda/AuthenticationCallbackHandlerTest.java:385]: Is this line making a difference? Looks like line 138 on this file has the same statement.

Comment (billrdunn) [reply oidc-api/src/test/java/uk/gov/di/authentication/oidc/lambda/AuthenticationCallbackHandlerTest.java:385]: Oh yes not sure why this got added

**Review by GHSwallow — APPROVED** [review]
lgtm!

**Pull request merged**